### ISSUE TITLE
Use mnemonic for process state (running, sleeping etc) on OS X and BSD 

### ISSUE THREAD

**Issue by sharvilshah** [facebook/osquery#1886]
The Linux implementation of in `processes` table outputs the process `state` column as a mnemonic:

```
       D    uninterruptible sleep (usually IO)
       R    running or runnable (on run queue)
       S    interruptible sleep (waiting for an event to complete)
       T    stopped, either by a job control signal or because it is being traced.
       W    paging (not valid since the 2.6.xx kernel)
       X    dead (should never be seen)
       Z    defunct ("zombie") process, terminated but not reaped by its parent.
```

 similar to `ps` output.

The current implementation of the same on OS X outputs an internal `int`, we should map those to the corresponding mnemonics.

**Comment by sharvilshah** [comment]
The correct process state codes and their corresponding strings on OS X are:

```
0	Zombie
1	Running
2	Stuck
3	Sleeping
4	idle
5	Stopped
6	Halted
7	Unknown
```

I'll try to get a PR up. While it's straightforward to have a lookup map, I want to investigate if there is a mach/libproc API that might return the string too.

Will also need to distinguish between `Stuck` and `Sleeping`

**Comment by theopolis** [comment]
Looks like Stuck is equivalent to D: https://nelsonslog.wordpress.com/2012/02/06/macos-top-stuck-means-uninterruptible/ (also seems D has been renamed to U).

But this says 4=Idle is uninterruptible: http://fxr.watson.org/fxr/source/bsd/sys/proc_info.h?v=xnu-2050.18.24#L212

Perhaps we should use htop's: https://github.com/hishamhm/htop/blob/master/darwin/DarwinProcess.c#L345-L353

That limits the selection to:
```
1 R Running
2 S Stopped
3 W Waiting
4 U Uninterruptible
5 H Halted 
``` 

No Zombie :(

**Comment by sharvilshah** [comment]
Thinking of going with how `ps` does it:

```c
#define STATE_MAX       7 
                
int     
mach_state_order(s, sleep_time)
int s;
long sleep_time;
{      
switch (s) {
case TH_STATE_RUNNING:      return(1);
case TH_STATE_UNINTERRUPTIBLE:
                                return(2);
    case TH_STATE_WAITING:      return((sleep_time > 20) ? 4 : 3);
    case TH_STATE_STOPPED:      return(5);
    case TH_STATE_HALTED:       return(6);  
    default:                    return(7); 
    }
}
                            /*01234567 */
char    mach_state_table[] = " RUSITH?";
```

https://opensource.apple.com/source/adv_cmds/adv_cmds-163/ps/tasks.c

And

```c
extern int mach_state_order();
void
state(k, ve)
	KINFO *k;
	VARENT *ve;
{
	struct extern_proc *p;
	int flag,j;
	char *cp;
	VAR *v;
	char buf[16];
	extern char mach_state_table[];

	v = ve->var;
	p = KI_PROC(k);
	flag = p->p_flag;
	cp = buf;

	if(!mflg ) {
	switch (p->p_stat) {

	case SSTOP:
		*cp = 'T';
		break;

	case SZOMB:
		*cp = 'Z';
		break;

	default:
		*cp = mach_state_table[k->state];
	}
```

from https://opensource.apple.com/source/adv_cmds/adv_cmds-163/ps/print.c

**Comment by sharvilshah** [comment]
Fixed in #1913
